DataSource/Driver/Doctrine/QueryBuilder/Filterer.php

```diff
@@ -147,74 +145,78 @@ protected function buildCondition(FilterCondition $filterCondition, $token, Quer
         $identifier = $this->uniqueNameToQueryFieldMap[$filterCondition->getFieldName()];
         $value = $filterCondition->getValue();
 
-        if ($filterMethod == FilterCondition::METHOD_STRING_LIKE) {
-            return $queryBuilder->expr()->like($identifier, $token);
-        } elseif ($filterMethod == FilterCondition::METHOD_STRING_EQ ||
-            $filterMethod == FilterCondition::METHOD_NUMERIC_EQ ||
-            $filterMethod == FilterCondition::METHOD_BOOLEAN ||
-            $filterMethod == FilterCondition::METHOD_DATETIME_EQ
-        ) {
-            return $queryBuilder->expr()->eq($identifier, $token);
-        } elseif ($filterMethod == FilterCondition::METHOD_NUMERIC_GT || $filterMethod == FilterCondition::METHOD_DATETIME_GT) {
-            return $queryBuilder->expr()->gt($identifier, $token);
-        } elseif ($filterMethod == FilterCondition::METHOD_NUMERIC_GTE || $filterMethod == FilterCondition::METHOD_DATETIME_GTE) {
-            return $queryBuilder->expr()->gte($identifier, $token);
-        } elseif ($filterMethod == FilterCondition::METHOD_NUMERIC_EQ || $filterMethod == FilterCondition::METHOD_DATETIME_EQ) {
-            return $queryBuilder->expr()->eq($identifier, $token);
-        } elseif ($filterMethod == FilterCondition::METHOD_NUMERIC_LTE || $filterMethod == FilterCondition::METHOD_DATETIME_LTE) {
-            return $queryBuilder->expr()->lte($identifier, $token);
-        } elseif ($filterMethod == FilterCondition::METHOD_NUMERIC_LT || $filterMethod == FilterCondition::METHOD_DATETIME_LT) {
-            return $queryBuilder->expr()->lt($identifier, $token);
-        } elseif ($filterMethod == FilterCondition::METHOD_STRING_NEQ || $filterMethod == FilterCondition::METHOD_NUMERIC_NEQ || $filterMethod == FilterCondition::METHOD_DATETIME_NEQ) {
-            return $queryBuilder->expr()->neq($identifier, $token);
-        } elseif ($filterMethod == FilterCondition::METHOD_IN) {
-            if (!is_array($value)) {
-                throw new \Exception('Only arrays can be arguments of a METHOD_IN filter');
-            }
-
-            if (count($filterCondition->getValue()) > 0) {
-                return $queryBuilder->expr()->in($identifier, $token);
-            }
-            // The array is empty, therefore this will always be "false". We use an always-false expression
-            // to emulate this without actually using an invalid empty array in the IN statement.
-            return '1=2';
-        } elseif ($filterMethod == FilterCondition::METHOD_IS_NULL) {
-            if ($filterCondition->getValue()) {
-                return $queryBuilder->expr()->isNull($identifier);
-            }
-
-            return $queryBuilder->expr()->isNotNull($identifier);
-        } else {
-            throw new \Exception("Unknown filtering method \"$filterMethod\" for column \"" . $filterCondition->getFieldName() . '"');
+        switch ($filterMethod) {
```

Comment:
- changes `if ... elseif ... elseif ... elseif ...` to switch/case (for me it looks simpler, more readable like this)
- removes duplicated condition check for `METHOD_NUMERIC_EQ` and `METHOD_DATETIME_EQ` (previously done twice, at lines 152 and 162)